internal/packager/init.go

```diff
@@ -198,7 +199,22 @@ func initFromComposeFile(name string, composeFile string) error {
 	if err == nil {
 		parseEnv(string(envRaw), settings)
 	}
-	keys, err := ExtractVariables(string(composeRaw))
+	compose := make(map[interface{}]interface{})
+	err = yaml.Unmarshal(composeRaw, compose)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse compose file")
+	}
+	version, ok := compose["version"]
```

Comment:
Could you extract the version check in a function?

Reply:
done and done, thanks!